src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -284,12 +287,15 @@ protected Query doToQuery(QueryShardContext context) {
         KNNMethodContext knnMethodContext = knnVectorFieldType.getKnnMethodContext();
         KNNEngine knnEngine = KNNEngine.DEFAULT;
         VectorDataType vectorDataType = knnVectorFieldType.getVectorDataType();
+        SpaceType spaceType = knnVectorFieldType.getSpaceType();
 
         if (fieldDimension == -1) {
             // If dimension is not set, the field uses a model and the information needs to be retrieved from there
+            assert spaceType == null;
```

Comment:
1. Putting assert here is dangerous as this will lead to exception messages not understood by users. Lets make sure that we are making sure proper exceptions are thrown here.

2. Due to this change the comment location have been moved. Lets ensure the comments is at right position.